### PR TITLE
Drawing Classifier: added docstrings and cleaned up code

### LINKED ISSUE #1544: Insufficient Python doc string for `tc.drawing_classifier.create`
```
import turicreate as tc
help(tc.drawing_classifier.create)
```
only yields
```
Help on function create in module turicreate.toolkits.drawing_classifier.drawing_classifier:

create(input_dataset, feature='drawing', target='label', pretrained_model_url=None, classes=None, batch_size=256, num_epochs=100, max_iterations=0, verbose=True, **kwargs)
    Create a :class:`DrawingClassifier` model.
```

Compare to what you get for e.g. `help(tc.image_classifier.create)`

### PR DESCRIPTION
**Pull request by shantanuchhabra**
Fixes #1544

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 1]
```diff
@@ -199,7 +274,88 @@ def _load_version(cls, state, version):
...
+        Returns
+        -------
+        out : string
+            A description of the DrawingClassifier.
+        """
+        return self.__repr__()
+
+    def __repr__(self):
+        """
+        Print a string description of the model when the model name is entered
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: `Print` -> `Returns`?

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 2]
```diff
@@ -310,149 +472,161 @@ def _predict_with_probabilities(self, input_dataset):
...
-        """ 
-        Docfix coming soon
+    def predict(self, data, verbose = True):
+        """
+        Predict on an SFrame or SArray of drawings, or on a single drawing.
+
+        Parameters
+        ----------
+        data : SFrame | SArray | tc.Image | list
+            The images on which to perform drawing classification.
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Probably `images` -> `image(s)`

File: src/unity/python/turicreate/toolkits/drawing_classifier/_sframe_loader.py [hunk 3]
```diff
@@ -16,10 +16,10 @@
 _TMP_COL_RANDOM_ORDER = '_random_order'
 
 # Encapsulates an SFrame, iterating over each row and returning an
-# (image, label, index) tuple.
+# (example, label, index) tuple.
 class _SFrameDataSource:
     def __init__(self, sframe, feature_column, target_column,
-                 load_labels=True, shuffle=True, samples=None):
```
[the remark this answers is not in the document]
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/_sframe_loader.py]: Thanks for sharing! My life was a lie.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 4]
```diff
@@ -154,6 +223,7 @@ def update_progress(cur_loss, iteration):
     state = {
         '_model': model,
         '_class_to_index': class_to_index,
+        'name': 'Drawing Classifier'
```
[the remark this answers is not in the document]
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Object Detector sort of does it. It adds a `model` parameter that is set to "darknet-yolo". Anyway, I followed the Image Classifier and removed this field.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 5]
```diff
@@ -42,6 +47,70 @@ def create(input_dataset, feature="drawing", target="label",
...
+    target : string
+        Name of the column containing the target. 'None' (the default)
+        indicates the column in `dataset` named "label" should be used as the
+        feature. The target column must be integer or string.
+
+    pretrained_model_url : string optional
+        A URL to the pretrained model that must be used for a warm start before
+        training.
+
+    classes : list optional
```
[the remark this answers is not in the document]
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Removed.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 6]
```diff
@@ -42,6 +47,70 @@ def create(input_dataset, feature="drawing", target="label",
...
+        Stroke-based drawing input must be in the following format:
+        Every drawing must be represented by a list of strokes, where each
+        stroke must be a list of points in the order in which they were drawn
+        on the canvas.
+        Each point must be a dictionary with two keys, "x" and "y", and their
+        respective values must be numerical, i.e. either integer or float.
+
+    target : string
+        Name of the column containing the target. 'None' (the default)
+        indicates the column in `dataset` named "label" should be used as the
```
[the remark this answers is not in the document]
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Done.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 7]
```diff
@@ -42,6 +47,70 @@ def create(input_dataset, feature="drawing", target="label",
...
+
+    Parameters
+    ----------
+    dataset : SFrame
+        Input data. The columns named by the ``feature`` and ``target``
+        parameters will be extracted for training the drawing classifier.
+
+    feature : string
+        Name of the column containing the input drawings. 'None' (the default)
+        indicates the column in `dataset` named "drawing" should be used as the
```
[the remark this answers is not in the document]
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Done.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (nickjong) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: The object detector is unusual, so you should copy patterns from it without understanding the context. In the case of OD, the "darknet-yolo" property indicates which pre-trained model was used for the warm start (guarding against the future when different model architectures are used)

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 8]
```diff
@@ -37,11 +42,77 @@ def _raise_error_if_not_drawing_classifier_input_sframe(
...
+    Parameters
+    ----------
+    dataset : SFrame
+        Input data. The columns named by the ``feature`` and ``target``
+        parameters will be extracted for training the drawing classifier.
+
+    target : string
+        Name of the column containing the target variable. The values in this
+        column must be of string or integer type. String target variables are
+        automatically mapped to integers in the order in which they are
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: This seems strange to me. I thought the other toolkits all sort the unique values to obtain the list of classes and therefore the mapping

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 9]
```diff
@@ -37,11 +42,77 @@ def _raise_error_if_not_drawing_classifier_input_sframe(
...
+        on the canvas.
+        Each point must be a dictionary with two keys, "x" and "y", and their
+        respective values must be numerical, i.e. either integer or float.
+
+    pretrained_model_url : string optional
+        A URL to the pretrained model that must be used for a warm start before
+        training.
+
+    batch_size: int optional
+        The number of images per training iteration. If not set, a default
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: It seems weird to use the term "iteration" here for a single training step, whereas below iteration refers to a pass through the data. (Some of this is constrained by the precedent set by some of the pre-deep-learning toolkits, which used "iteration" for what NN practitioners probably call "epoch")

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 8 above)
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Thanks for catching my typo. There is no `classes` parameter anymore so this doesn't make sense. We do sort the unique values behind the scenes to obtain a mapping.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 9 above)
Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Great catch, thank you. I'll change the wording to "training step", which is clearer and avoids confusion with `max_iterations` where iteration is supposed to mean an epoch.

File: src/unity/python/turicreate/toolkits/drawing_classifier/_sframe_loader.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickjong) [reply src/unity/python/turicreate/toolkits/drawing_classifier/_sframe_loader.py]: FWIW, we like the spaces around `=` on the C++ side :smile:

**Pull request merged**